cmd/xl-v1-list-objects-heal.go

```diff
@@ -205,3 +206,252 @@ func (xl xlObjects) ListObjectsHeal(bucket, prefix, marker, delimiter string, ma
 	// Return error at the end.
 	return ListObjectsInfo{}, toObjectErr(err, bucket, prefix)
 }
+
+// ListUploadsHeal - lists ongoing multipart uploads that require
+// healing in one or more disks.
+func (xl xlObjects) ListUploadsHeal(bucket, prefix, marker, uploadIDMarker,
+	delimiter string, maxUploads int) (ListMultipartsInfo, error) {
+
+	if err := checkListObjsArgs(bucket, prefix, marker, delimiter, xl); err != nil {
+		return ListMultipartsInfo{}, err
+	}
+
+	// With max uploads of zero we have reached eof, return right here.
+	if maxUploads == 0 {
+		return ListMultipartsInfo{}, nil
+	}
+
+	// For delimiter and prefix as '/' we do not list anything at all
+	// since according to s3 spec we stop at the 'delimiter' along
+	// with the prefix. On a flat namespace with 'prefix' as '/'
+	// we don't have any entries, since all the keys are of form 'keyName/...'
+	if delimiter == slashSeparator && prefix == slashSeparator {
+		return ListMultipartsInfo{}, nil
+	}
+
+	// If maxUploads is out of bounds reset it to maxUploadsList
+	if maxUploads < 0 || maxUploads > maxUploadsList {
+		maxUploads = maxUploadsList
+	}
+
+	// Initiate a list operation, if successful filter and return quickly.
```

Comment:
This code snippet was 'inspired' from listObjectsHeal. This comment needs to be modified since it doesn't make sense after the modifications I did to the version.